src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/FormFileValueProvider.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// An <see cref="IValueProvider"/> adapter for data stored in an <see cref="IFormFileCollection"/>.
+    /// Does not actually read the form collection, but provides prefix matches.
+    /// </summary>
+    internal class FormFileValueProvider : IEnumerableValueProvider
```

Comment:
Rate this on a scale of 😸 to 🤢

Reply:
It should probably be public so people can remove